src/appleseed/renderer/modeling/frame/frame.cpp

```diff
@@ -1071,29 +1038,48 @@ namespace
 
         try
         {
-            if (extension == ".exr")
-            {
-                write_exr_image(
-                    bf_file_path,
-                    image,
-                    image_attributes);
+            bool high_dynamic_range_format = false;
+            if (extension == ".exr" || extension == ".tiff" || extension == ".tif" || extension == ".hdr")
+                high_dynamic_range_format = true;            
+
+            create_parent_directories(bf_file_path);
+
+            const std::string filename = bf_file_path.string();
+            GenericImageFileWriter writer(filename.c_str());
+
+            // Add image to writer
+            Image *transformed_image;
+            if (!high_dynamic_range_format && image.properties().m_channel_count == 4) {
+                transformed_image = new Image(image);
+                transform_to_srgb(transformed_image);
+                writer.append_image(transformed_image);
             }
-            else if (extension == ".png")
-            {
-                write_png_image(
-                    bf_file_path,
-                    image,
-                    image_attributes);
+            else if (extension == ".hdr") {
+                // .hdr file only support 3 channel
+                transformed_image = new Image(image, 3);
+                writer.append_image(transformed_image);
             }
-            else
-            {
+            else if (high_dynamic_range_format){
+                transformed_image = new Image(image);
+                writer.append_image(transformed_image);
+            }
+            else {
                 RENDERER_LOG_ERROR(
-                    "failed to write image file %s for frame \"%s\": unsupported image format.",
-                    bf_file_path.string().c_str(),
-                    frame.get_path().c_str());
-
+                    "failed to write image file %s: unsupport image format.",
```

Comment:
Why did you change this? Please revert.

Reply:
I change this because you want to automatically handle the different formats using OIIO.
I just put the function `write_png_image` and `write_exr_image` in `'write_image`.

Reply:
You're still introducing a typo here: *unsupport* -> *unsupported*

Reply:
Thanks, done.